doctr/models/recognition/master/pytorch.py

```diff
@@ -0,0 +1,290 @@
+            nn.MaxPool2d(2),
+            # conv_2x
+            *resnet_stage(128, 256, num_blocks=1),
+            MAGC(inplanes=256, headers=headers, att_scale=True),
+            *conv_sequence_pt(256, 256, relu=True, bn=True, kernel_size=3, padding=1),
+            nn.MaxPool2d(2),
+            # conv_3x
+            *resnet_stage(256, 512, num_blocks=2),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+            nn.MaxPool2d((2, 1)),
+            # conv_4x
+            *resnet_stage(512, 512, num_blocks=5),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+            # conv_5x
+            *resnet_stage(512, 512, num_blocks=3),
+            MAGC(inplanes=512, headers=headers, att_scale=True),
+            *conv_sequence_pt(512, 512, relu=True, bn=True, kernel_size=3, padding=1),
+        ]
+        super().__init__(*_layers)
+
+
+class MASTER(_MASTER, nn.Module):
+
+    """Implements MASTER as described in paper: <https://arxiv.org/pdf/1910.02562.pdf>`_.
+    Implementation based on the official Pytorch implementation: <https://github.com/wenwenyu/MASTER-pytorch>`_.
+
+    Args:
+        vocab: vocabulary, (without EOS, SOS, PAD)
+        d_model: d parameter for the transformer decoder
+        headers: headers for the MAGC module
+        dff: depth of the pointwise feed-forward layer
+        num_heads: number of heads for the mutli-head attention module
+        num_layers: number of decoder layers to stack
+        max_length: maximum length of character sequence handled by the model
+        input_size: size of the image inputs
+    """
+
+    def __init__(
+        self,
+        vocab: str,
+        d_model: int = 512,
+        headers: int = 1,
+        dff: int = 2048,
+        num_heads: int = 8,
+        num_layers: int = 3,
+        max_length: int = 50,
+        input_shape: Tuple[int, int, int] = (3, 48, 160),
+        cfg: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__()
+
+        self.max_length = max_length
+        self.vocab = vocab
+        self.cfg = cfg
+        self.vocab_size = len(vocab)
+        self.num_heads = num_heads
+
+        self.feature_extractor = MAGCResnet(headers=headers)
+        self.seq_embedding = nn.Embedding(self.vocab_size + 1, d_model)  # One additional class for EOS
+
+        self.decoder = Decoder(
+            num_layers=num_layers,
+            d_model=d_model,
+            num_heads=num_heads,
+            dff=dff,
+            vocab_size=self.vocab_size,
+            maximum_position_encoding=max_length,
+        )
+        self.feature_pe = positional_encoding(input_shape[1] * input_shape[2], d_model)
+        self.linear = nn.Linear(d_model, self.vocab_size + 1)
+        nn.init.kaiming_uniform_(self.linear.weight)
```

Comment:
In PyTorch, the default initialization is already well done: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/linear.py#L92

Apart from if you're using a custom activation afterwards, I don't suggest changing that